Userland/Libraries/LibC/locale.cpp

```diff
@@ -45,11 +45,16 @@ static struct lconv default_locale = {
     default_empty_value
 };
 
-char* setlocale(int, char const*)
+char* setlocale(int, char const* locale)
 {
-    static char locale[2];
-    memcpy(locale, "C", 2);
-    return locale;
+    // If we get a null pointer, return the current locale as per POSIX spec.
+    if (locale == nullptr)
+        return strdup("C");
```

Comment:
This patch has the right idea, but the implementation leaks a 2 byte string every time setlocale is called.

Per Posix, we are allowed to return a pointer to static storage.

> The returned string pointer might be invalidated or the string content might be overwritten by a subsequent call to setlocale().  The returned pointer might also be invalidated if the calling thread is terminated.

I would prefer to keep that property of the original implementation and return a pointer to a static buffer holding "C"